rel/expr_cond_pattern_control_var.go

```diff
@@ -48,6 +52,19 @@ func (expr CondPatternControlVarExpr) Eval(scope Scope) (Value, error) {
 		return nil, WrapContext(err, expr.controlVarExpr, scope)
 	}
 
+	// cond is order dependent. It tries patterns from left to right and stops at the first match.
+	// If the first pattern is ident, it can't reach the subsequent patterns.
+	// So print out an error message to remind user if the order of cond patterns is correct.
+	if len(expr.conditionPairs) > 1 {
+		if ident, isIdent := expr.conditionPairs[0].expr.(IdentExpr); isIdent {
+			loggingOnce.Do(func() {
+				log.Error(context.Background(),
+					errors.Errorf("the first cond pattern is %s, it is causing subsequent patterns are unreachable, "+
+						"please make sure if the order of pattern is correct.", ident))
+			})
```

Comment:
Are you able to test that an error was logged?

Reply:
Yes, it was logged.
```
@> cond [1, 2] {k:k, [x, y]: x + y}
2020-06-22T12:31:42.280099+08:00 error_message=the first cond pattern is k, it is causing subsequent patterns are unreachable, please make sure if the order of pattern is correct. INFO 
[1, 2]
@>  
```

Reply:
Sorry, I mean are you able to write a unit test to verify that it is logged, to prevent regressions?

But also don't bother since Marcelo says not to log at all 🙃

Reply:
Will fix in #484